sources/ghostdog/src/main.rs

```diff
@@ -125,6 +168,51 @@ fn parse_device_name(device_info: &[u8], path: String) -> Result<String> {
         .to_string())
 }
 
+/// Read a file into a SupportedDevicesConfiguration Enum
+fn read_supported_devices_file(path: PathBuf) -> Result<SupportedDevicesConfiguration> {
+    let mut supported_devices_file =
+        fs::File::open(&path).context(error::OpenFileSnafu { path: path.clone() })?;
+    let mut supported_devices_str = String::new();
+    supported_devices_file
+        .read_to_string(&mut supported_devices_str)
+        .context(error::ReadFileSnafu { path: path.clone() })?;
+    let device_configuration: SupportedDevicesConfiguration =
+        serde_json::from_str(supported_devices_str.as_str())
+            .context(error::ParseGpuDevicesFileSnafu {})?;
+    Ok(device_configuration)
+}
+
+/// Search the Open GPU Supported Devices File to determine if the Open GPU Driver should be used based upon PCI devices present
+fn find_preferred_driver() -> Result<String> {
+    let open_gpu_devices = read_supported_devices_file(OPEN_GPU_SUPPORTED_DEVICES_PATH.into())?;
+    let list_input = pciclient::ListDevicesParam::builder()
+        .vendor(NVIDIA_VENDOR_ID)
+        .build();
+    let present_devices =
+        pciclient::list_devices(list_input).context(error::ListPciDevicesSnafu {})?;
+
+    // If there a multiple devices with the same ID, dedup them to minimize iterations
+    let mut unique_ids = present_devices
+        .iter()
+        .map(|x| format!("0x{}", x.device().to_uppercase()).clone());
+
+    let (device_list, driver_choice) = match &open_gpu_devices {
+        SupportedDevicesConfiguration::OpenGpu(ref device_list) => (
+            device_list
+                .iter()
+                .map(|x| &x.device_id)
+                .collect::<HashSet<_>>(),
+            "open-gpu",
+        ),
+    };
+
+    if unique_ids.any(|input_device| device_list.contains(&input_device)) {
+        Ok(driver_choice.to_string())
```

Comment:
nit: I think it'd be more clear to use the string here and not bother with the tuple:
```suggestion
        Ok("open-gpu".to_string())
```

Reply:
I spoke with @cbgbt offline about this approach. The rationale here was to make this ready to take other driver choices if needed in the future since we are using a single variant of an enum. Nonetheless, we don't have other variants yet so we can simplify the code as it is today.